NanaZip.UI.Classic/SevenZip/CPP/7zip/UI/FileManager/PanelFolderChange.cpp

```diff
@@ -72,10 +72,73 @@ static bool DoesNameContainWildcard_SkipRoot(const UString &path)
 {
   return DoesNameContainWildcard(path.Ptr(NName::GetRootPrefixSize(path)));
 }
+// **************** NanaZip Modification Start **************
+bool HasDriveLetter(const UString& path)
+{
+    return path.Len() >= 2 &&
+        ((path[0] >= L'A' && path[0] <= L'Z') ||
+        (path[0] >= L'a' && path[0] <= L'z')) &&
+        path[1] == L':';
+}
 
+UString ExpandFirstEnvironmentInPath(const UString &path)
+{
+    int firstPercent = path.Find(L'%');
+    if (firstPercent == std::wstring::npos)
+    {
+        return path;
+    }
+    int secondPercent = path.Find(L'%', firstPercent + 1);
+    if (secondPercent == std::wstring::npos)
+    {
+        return path;
+    }
+    UString environmentName;
+    for (int i = firstPercent; i <= secondPercent; ++i)
+    {    
+        environmentName += path[i];
+    }
+    wchar_t buffer[MAX_PATH * 4] = {};
+    DWORD bufSize = ::ExpandEnvironmentStringsW(
+        environmentName.Ptr(),
+        buffer, 
+        ARRAYSIZE(buffer));
+    UString expandedFirstEvironment;
+    if (bufSize == 0 || bufSize > ARRAYSIZE(buffer))
+    {
+        expandedFirstEvironment = environmentName;
+    }
+    else
+    {
+        expandedFirstEvironment = buffer;
+    }
+    UString expandedPath;
+    if (firstPercent > 0)
+    {
+        expandedPath += path.Left(firstPercent);
+    }
+    expandedPath += expandedFirstEvironment;
+    if (secondPercent + 1 <(int) path.Len())
+    {
+        expandedPath += path.Ptr(secondPercent + 1);
+    }
+    return expandedPath;
+}
+// **************** NanaZip Modification End ****************
```

Comment:
Add an empty line after NanaZip Modification End in this scenario for better git diff result.